storage/scratchpad/src/sparse_merkle/mod.rs

```diff
@@ -101,156 +101,55 @@ use aptos_types::{
 };
 use std::{
     collections::{BTreeMap, HashMap},
-    sync::{Arc, MutexGuard, Weak},
+    sync::Arc,
 };
 use thiserror::Error;
 
 type NodePosition = bitvec::vec::BitVec<u8, bitvec::order::Msb0>;
 const BITS_IN_NIBBLE: usize = 4;
 const BITS_IN_BYTE: usize = 8;
 
-/// To help finding the oldest ancestor of any SMT, a branch tracker is created each time
-/// the chain of SMTs forked (two or more SMTs updating the same parent).
-#[derive(Debug)]
```

Comment:
how are we going to track forks ?

Reply:
"children"
The branch tracker tracks "backwards" to the ancestors, see the pr mentioned in the summary.